### PR TITLE
In PeoplePicker, we can now use arrow keys to move cursor around #28655

### LINKED ISSUE #28655: [Bug]: In PeoplePicker, can not use arrow keys to move cursor around
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 28.25 GB / 47.71 GB
  Browsers:
    Edge: Spartan (44.22621.1992.0), Chromium (114.0.1823.86)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/peoplepicker

## Actual Behavior
In the input box of PeoplePicker, after typing something, pressing left or right arrow key clears the typed content. 
However, if backspece has been used to delete characters, then pressing left or right arrow key does not clear typed content and correctly moves cursor to the left / right.

## Expected Behavior
User should be able to use arrow key to move cursor around within input box. 

### Logs

_No response_

### Requested priority

Low

### Products/sites affected

PowerApps

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by azizChebbi**
## Previous Behavior

The problem was basically with the AutoFill component, pressing the left/right key was clearing the input field rather than moving the cursor in the right direction.

## New Behavior

When pressing the left/right keys the cursor goes to the right direction, which is the expected behavior.

## Related Issue(s)

In PeoplePicker, can not use arrow keys to move cursor around 
- Fixes #28655

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/78808a4e906580cd2b7a946b8be359eb548206bc'>78808a4e906580cd2b7a946b8be359eb548206bc</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react/src/components/Autofill/Autofill.tsx [hunk 1]
```diff
@@ -255,7 +255,9 @@ export class Autofill extends React.Component<IAutofillProps, IAutofillState> im
         case KeyCodes.left:
```
Comment (fabricteam) [inline packages/react/src/components/Autofill/Autofill.tsx:255]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28975&commitId=fdad633015bfe9aa51b86ca163d3ec15453479c7&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-VerticalBarChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis.default.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by azizChebbi** [comment]
> @azizChebbi please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Comment by azizChebbi** [comment]
@microsoft-github-policy-service agree

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**